packages/@react-aria/toast/src/useToastRegion.ts

```diff
@@ -49,6 +49,32 @@ export function useToastRegion<T>(props: AriaToastRegionProps, state: ToastState
     onHoverEnd: state.resumeAll
   });
 
+  let prevToastCount = useRef(state.visibleToasts.length);
+  useEffect(() => {
+    // Resume timers if the user's pointer left the region due to a toast being removed and the region shrinking.
+    // Waits until the next pointermove after a toast is removed.
+    let onPointerMove = (e: PointerEvent) => {
+      if (!ref.current) {
+        document.removeEventListener('pointermove', onPointerMove);
+        return;
+      }
+      let regionRect = ref.current.getBoundingClientRect();
+      const isPointerOverRegion = e.clientX >= regionRect.left && e.clientX <= regionRect.right && e.clientY >= regionRect.top && e.clientY <= regionRect.bottom;
+      if (!isPointerOverRegion) {
+        state.resumeAll();
+      }
+      document.removeEventListener('pointermove', onPointerMove);
+    };
+
+    if (state.visibleToasts.length < prevToastCount.current && state.visibleToasts.length > 0) {
+      document.addEventListener('pointermove', onPointerMove);
```

Comment:
if we use `once`, then it simplifies some of the cleanup
```suggestion
      document.addEventListener('pointermove', onPointerMove, {once: true});
```